### PR TITLE
Validate the dwell_time parameter's datatype 

### PR DESCRIPTION
**Pull request by subinsaji**
Fixes [#895](https://github.com/DiamondLightSource/hyperion/issues/895) in Hyperion

- Added pydantic validator for `dwell_time` in **fast_grid_scan.py** and  checks if `dwell_time` is an integer value
- Parametrized tests with multiple dwell times using the logic from **fast_grid_scan.py** in **test_gridscan.py**

### Instructions to reviewer on how to test:
1. Run Tests
### Checks for reviewer
- [x] Would the PR title make sense to a scientist on a set of release notes
- [x] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

File: src/dodal/devices/fast_grid_scan.py [hunk 1]
```diff
@@ -91,6 +91,15 @@ def _get_y_axis(cls, y_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
     def _get_z_axis(cls, z_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
         return GridAxis(values["z2_start"], values["z_step_size"], values["z_steps"])
 
+    @validator("dwell_time", always=True)
+    def non_ineger_dwell_time(cls, dwell_time: float):
```
Comment (DominicOram) [inline src/dodal/devices/fast_grid_scan.py:95]: Should: You're missing a `t`in `integer`

File: src/dodal/devices/fast_grid_scan.py [hunk 2]
```diff
@@ -91,6 +91,15 @@ def _get_y_axis(cls, y_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
...
+        nums_a = dwell_time * 1000
+        nums_b = np.floor(dwell_time * 1000)
```
Comment (DominicOram) [inline src/dodal/devices/fast_grid_scan.py:97]: Nit: `nums_a` and `nums_b` aren't very descriptive names, could you come up with better ones?

File: src/dodal/devices/fast_grid_scan.py (hunk 1 above)
Comment (DominicOram) [inline src/dodal/devices/fast_grid_scan.py:95]: Should: Could you add a type annotation to show what type the function returns?

File: src/dodal/devices/fast_grid_scan.py [hunk 3]
```diff
@@ -91,6 +91,15 @@ def _get_y_axis(cls, y_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
...
+        num_is_close = np.isclose(nums_a, nums_b, rtol=1e-1)
+        if not num_is_close:
+            raise ValueError("Non integer value")
```
Comment (DominicOram) [inline src/dodal/devices/fast_grid_scan.py:100]: Should: This isn't a very descriptive exception, would be good to say what's not an integer and what value it is.

File: tests/devices/unit_tests/test_gridscan.py [hunk 4]
```diff
@@ -337,3 +337,31 @@ def test_given_x_y_z_steps_when_full_number_calculated_then_answer_is_as_expecte
...
+        (0.0099, True),
+        (0.0059, False),
+        (0.0004, False),
+        (0.0009, False),
+        (0.00044, False),
+        (0.00099, False),
+        (0.00001, False),
+        (0.00009, False),
+        (0.000001, False),
+        (0.000009, False),
```
Comment (DominicOram) [inline tests/devices/unit_tests/test_gridscan.py:358]: Nice lot of test coverage, thanks!

**Review by DominicOram — DISMISSED** [review]
Looks mostly there, thanks. Some comments in code. Also, could you fix the CI linting and tests?

**Review by olliesilvester — APPROVED** [review]
This looks good to me! I just changed the name of the PR so it makes more sense on release notes

**Pull request merged**